src/main/java/org/openlmis/integration/dhis2/service/PeriodGeneratorService.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * This program is part of the OpenLMIS logistics management information system platform software.
+ * Copyright © 2017 VillageReach
+ *
+ * This program is free software: you can redistribute it and/or modify it under the terms
+ * of the GNU Affero General Public License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero General Public License for more details. You should have received a copy of
+ * the GNU Affero General Public License along with this program. If not, see
+ * http://www.gnu.org/licenses.  For additional information contact info@OpenLMIS.org.
+ */
+
+package org.openlmis.integration.dhis2.service;
+
+import java.time.Clock;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import org.openlmis.integration.dhis2.domain.enumerator.DhisPeriod;
+import org.springframework.data.util.Pair;
+import org.springframework.stereotype.Service;
+
+@Service
+public class PeriodGeneratorService {
+
+  private final Clock clock;
+
+  public PeriodGeneratorService() {
+    this.clock = Clock.systemDefaultZone();
```

Comment:
I think @sradziszewski was doing something similar with Clock - was there a bean factory method created for Clock, so this can be Autowired? Or am I remembering wrong?

Reply:
I haven't used the Clock class in the OLMIS project yet, but I see that there are bean definitions for this class in the repositories, e.g. 
[referencedata](https://github.com/OpenLMIS/openlmis-referencedata/blob/master/src/main/java/org/openlmis/referencedata/Application.java#L221)
[stockmanagement](https://github.com/OpenLMIS/openlmis-stockmanagement/blob/master/src/main/java/org/openlmis/stockmanagement/Application.java#L71)

Reply:
I added Clock Bean to Application.java and autowired it in PeriodGeneratorService